### DISCUSSION TITLE
Increase dashboard size limit

### DISCUSSION THREAD

**Discussion by Yattabyte** [apache/superset#13629] [Q&A / Help] [answered]
Hi there, I'm currently evaluating superset for my job, and I'm trying to create rather large dashboards, and have seem to be nearing the maximum number of plots and/or elements I can fit in it:
![image](https://user-images.githubusercontent.com/11085106/111176662-88f08a80-8577-11eb-944f-94b7b9944dcd.png)

I've tried digging through the code to find a variable that controls this limit, and added the following to my config.py file:
`SUPERSET_DASHBOARD_POSITION_DATA_LIMIT = 131072`

and re-ran docker compose
`docker-compose -f docker-compose-non-dev.yml up`

However I'm still virtually at max capacity.

Any insights?

**Comment by NikosSpanos** [comment]
Any solution that worked for this issue. Currently searching for a solution on this too.

**Comment by mattitoo** [comment]
Did you enter it into the correct file? If you are using Docker, you should put your changes into `docker/pythonpath/superset_config_docker.py`(you can take the example file in that directory and copy and rename it)

**Reply by NikosSpanos** [reply]
@mattitoo In my ```superset_config.py``` file under ```docker``` directory, the value ```SUPERSET_DASHBOARD_POSITION_DATA_LIMIT``` does not exist. It exists on the ```config.py``` file under directory ```superset```. Shall I copy the default value ```SUPERSET_DASHBOARD_POSITION_DATA_LIMIT = 65535``` and paste it in ```superset_config.py``` for the change to take effect?

**Comment by mattitoo** [comment, marked as answer]
In principle yes, but save it into superset_config_docker.py (which does not exist by default, but can be created from the example file). The values in this file will override the values in config.py.

**Reply by NikosSpanos** [reply]
OK, I will try this approach and keep you updated. Does it happen to know indicative values for this parameter? For example, SUPERSET_DASHBOARD_POSITION_DATA_LIMIT = 65000 is MegaBytes of data? Is it RAM memory allocated?

**Reply by nytai** [reply]
it's the number of chars. the default 65,535 is the MySQL max for TEXT columns.

**Comment by NikosSpanos** [comment]
@mattitoo  I can confirm that the approach worked. I added the value ```SUPERSET_DASHBOARD_POSITION_DATA_LIMIT = 131070``` in the ```superset_config.py``` and the message is ```Your Dashboard is near the size limit``` vanished.

**Comment by axoxymous-coder** [comment]
Hi,

Can someone help on how I change the value of `SUPERSET_DASHBOARD_POSITION_DATA_LIMIT` in Kubernetes deployments? We tried to set this value in `values.yaml` under section `extraConfigs`, but seems its not being set.

Any help will be highly appreciated.

Thanks.

**Reply by SamraHanifCareem** [reply]
You should ass it under configOverrides 
```
configOverrides:
feature_flag: |
SUPERSET_DASHBOARD_POSITION_DATA_LIMIT: data_limit
```

**Reply by Fineken** [reply]
Was there any way to solve this problem? Specifying the parameter did not help

**Comment by dolph** [comment]
Superset is storing JSON into a `TEXT` column. Increasing the value of `SUPERSET_DASHBOARD_POSITION_DATA_LIMIT` does not increase the actual size limit of the `TEXT` column - it just circumvents the (useful) warnings about dashboard size limits in the UI. It should not be a configurable value, IMO.

**Reply by sfirke** [reply]
I see how that would be problematic if you were running an old version of MySQL as your metadata db. But on a modern version of Postgres you can safely bump this limit and store larger dashboards. See https://github.com/apache/superset/issues/15169#issuecomment-1011902952 for more context.

**Comment by naveen-flow** [comment]
can someone help me !
Is it possible to expand the filter box size, make it auto fit with the text character size in its drop-downs in the dashboard
